+xpra.org

sebst opened the pull request:
Some help is needed with this PR.

- Is there a documentation on the replace tokens in the yaml (e.g. what would be the full version of an installed package? `{{deps.cairographics.org.pycairo.version.full}}`? How are `xy.zz/abc` package names handled?)
- Is there a token for `/home/user/.pkgx`?
- How could we handle Python packages (the mv/ln commands borrowed from `pycairo` itself do not seem to work here

Related: #7292 , #7290

sebst committed:
+xpra.org (wip)

sebst committed:
Merge branch 'main' into new/xpra.org

jhheider commented:
> Some help is needed with this PR.
> 
> * Is there a documentation on the replace tokens in the yaml (e.g. what would be the full version of an installed package? `{{deps.cairographics.org.pycairo.version.full}}`? How are `xy.zz/abc` package names handled?)

`{{deps.cairographics.org/pycairo.version}}`

> * Is there a token for `/home/user/.pkgx`?

`{{pkgx.prefix}}`

> * How could we handle Python packages (the mv/ln commands borrowed from `pycairo` itself do not seem to work here

multiple ways to deal with python: the "simplest" for a python "program" is `bkpyvenv`, as seen here: https://github.com/pkgxdev/pantry/blob/main/projects/mesonbuild.com/package.yml

> 
> Related: #7291 , #7290

sebst committed:
wip: replace tokens, use bkpyvenv

sebst committed:
+xpra.org (wip)

sebst committed:
+xpra.org (wip)

sebst commented:
> `{{deps.cairographics.org/pycairo.version}}`
> `{{pkgx.prefix}}`

Is there a list of tokens? I couldn't find documentation about it.

sebst committed:
+xpra.org (wip)

sebst commented:
> > `{{deps.cairographics.org/pycairo.version}}`
> > `{{pkgx.prefix}}`
> 
> Is there a list of tokens? I couldn't find documentation about it.

Found it. For ref: https://github.com/pkgxdev/pantry/wiki/Structure-of-a-%60package.yml%60#a-note-about-moustaches

sebst committed:
+xpra.org (wip, cleanup)

sebst committed:
+xpra.org: cleanup

sebst committed:
+xpra.org: re-organized script

sebst committed:
Merge branch 'main' into new/xpra.org

jhheider committed:
trying to simplify things

removed redundant deps (all runtime deps are build deps), tried to clean up the py3cairo.pc logic to be simpler. if i messed something up, i'll get it fixed. use the moustaches throughout.

jhheider committed:
fix yaml formatting

jhheider committed:
dropped the subpath
